sql-plugin/src/main/scala/com/nvidia/spark/rapids/RmmRapidsRetryIterator.scala

```diff
@@ -580,9 +590,14 @@ object RmmRapidsRetryIterator extends Logging {
             lastException = ex
 
             if (!topLevelIsRetry && !causedByRetry) {
-              // we want to throw early here, since we got an exception
-              // we were not prepared to handle
-              throw lastException
+              if (isOrCausedByColumnSizeOverflow(ex)) {
```

Comment:
We need to update `withRestoreOnRetry` for this new set of exceptions too.

Reply:
I think I've handled it now.
Since it's not possible to split and retry in these functions, I'm restoring and throwing the original exception.